provider/mediaconvert/mediaconvert.go

```diff
@@ -0,0 +1,376 @@
+		presetID, ok := output.Preset.ProviderMapping[Name]
+		if !ok {
+			return nil, provider.ErrPresetMapNotFound
+		}
+
+		wg.Add(1)
+		go p.makeGetPresetRequest(presetID, presetCh, &wg)
+	}
+
+	go func() {
+		wg.Wait()
+		close(presetCh)
+	}()
+
+	for res := range presetCh {
+		if res.err != nil {
+			return nil, fmt.Errorf("error getting preset info: %s", res.err)
+		}
+
+		presets[res.presetID] = res.preset
+	}
+
+	outputGroups := map[mediaconvert.ContainerType][]db.TranscodeOutput{}
+	for _, output := range job.Outputs {
+		presetID, ok := output.Preset.ProviderMapping[Name]
+		if !ok {
+			return nil, provider.ErrPresetMapNotFound
+		}
+
+		preset, ok := presets[presetID]
+		if !ok {
+			return nil, errors.New("mediaconvert preset not found in preset results")
+		}
+
+		container := preset.Settings.ContainerSettings.Container
+		outputGroups[container] = append(outputGroups[container], output)
+	}
+
+	mcOutputGroups := []mediaconvert.OutputGroup{}
+	for container, outputs := range outputGroups {
+		mcOutputGroup := mediaconvert.OutputGroup{}
+
+		var mcOutputs []mediaconvert.Output
+		for _, output := range outputs {
+			presetID, ok := output.Preset.ProviderMapping[Name]
+			if !ok {
+				return nil, provider.ErrPresetMapNotFound
+			}
+
+			rawExtension := path.Ext(output.FileName)
+			filename := strings.Replace(path.Base(output.FileName), rawExtension, "", 1)
+			extension := strings.Replace(rawExtension, ".", "", -1)
+
+			mcOutputs = append(mcOutputs, mediaconvert.Output{
+				Preset:       aws.String(presetID),
+				NameModifier: aws.String(filename),
+				Extension:    aws.String(extension),
+			})
+		}
+		mcOutputGroup.Outputs = mcOutputs
+
+		destination := destinationPathFrom(p.cfg.Destination, job.ID)
+
+		switch container {
+		case mediaconvert.ContainerTypeM3u8:
+			mcOutputGroup.OutputGroupSettings = &mediaconvert.OutputGroupSettings{
+				Type: mediaconvert.OutputGroupTypeHlsGroupSettings,
+				HlsGroupSettings: &mediaconvert.HlsGroupSettings{
+					Destination:            aws.String(destination),
+					SegmentLength:          aws.Int64(int64(job.StreamingParams.SegmentDuration)),
+					MinSegmentLength:       aws.Int64(0),
+					DirectoryStructure:     mediaconvert.HlsDirectoryStructureSingleDirectory,
+					ManifestDurationFormat: mediaconvert.HlsManifestDurationFormatFloatingPoint,
+					OutputSelection:        mediaconvert.HlsOutputSelectionManifestsAndSegments,
+					SegmentControl:         mediaconvert.HlsSegmentControlSegmentedFiles,
+				},
+			}
+		case mediaconvert.ContainerTypeMp4:
+			mcOutputGroup.OutputGroupSettings = &mediaconvert.OutputGroupSettings{
+				Type: mediaconvert.OutputGroupTypeFileGroupSettings,
+				FileGroupSettings: &mediaconvert.FileGroupSettings{
+					Destination: aws.String(destination),
+				},
+			}
+		default:
+			return nil, fmt.Errorf("container %s is not yet supported with mediaconvert", string(container))
+		}
+
+		mcOutputGroups = append(mcOutputGroups, mcOutputGroup)
+	}
```

Comment:
Same here, can probably create a separate function to build the output groups.

Reply:
Will do.